base/src/test/engineering/test_complex.rs

```diff
@@ -79,7 +79,11 @@ fn fn_imcos() {
 
     model.evaluate();
 
-    assert_eq!(model._get_text("A1"), "-6.58066304055116+7.58155274274654i");
+    let value = model._get_text("A1");
+    assert!(
+        value == "-6.58066304055116+7.58155274274654i"
+            || value == "-6.58066304055116+7.58155274274655i"
```

Comment:
I think the correct fix would be to check on an approximation of the real and imaginary parts. This fix could get out of hand..

For an example see:

https://github.com/ironcalc/IronCalc/blob/main/base/src/test/engineering/test_complex.rs#L144-L162

Reply:
Yikes!  I like the updated version with the approximation even less.

https://github.com/ironcalc/IronCalc/pull/19/commits/e3c20b2ceaecb3d12d5f174d38b01480f5ece452

Reply:
Hi, sorry for the late reply, the way to do that would be to test the function `=IMAGINARY(IMCOS(x))`

Or testing the absolute value is less than `f64::EPS`

Reply:
I think I'm missing something here when converting from a string to a float.  Updating to the following:

```rust
#[test]
fn fn_imcos() {
    let mut model = new_empty_model();
    model._set("A1", r#"0.11212+1.234523I"#);
    model._set("A2", r#"=IMABS(IMCOS(3)-A1)"#);

    model.evaluate();

    let result_str = model._get_text("A2");
    let float_value = match result_str.parse::<f64>() {
        Ok(value) => value,
        Err(err) => {
            panic!("{}", err);
        }
    };

    assert!(float_value < f64::EPSILON);
}
```

Results in:

```sh
running 1 test
thread 'test::engineering::test_complex::fn_imcos' panicked at base/src/test/engineering/test_complex.rs:87:13:
invalid float literal
```

Reply:
What is happening is that `result_str` is `#VALUE!`. You can't use `+` and `-` to add or subtract complex numbers. In other words the arithmetic operators are not overloaded. Yo have to use functions like `IMADD` or `IMSUB`.

See the list of Excel functions here:
https://support.microsoft.com/en-au/office/excel-functions-by-category-5f91f4e9-7b42-46d2-9bd1-63f26a86c0eb

also you can test in the playground if it works:
https://playground.ironcalc.com/